### PR TITLE
add debug logging to chat

### PR DESCRIPTION
**Pull request by makelinux**
To assist debugging and reverse engineering.

Samples of messages:
```
DEBUG 2024-06-11 13:59:39,057 lab.py:697: chat 
DEBUG 2024-06-11 15:43:01,308 chat.py:482: chat_cli creating OpenAI client
DEBUG 2024-06-11 15:43:01,351 chat.py:81: __init__ ConsoleChatBot
DEBUG 2024-06-11 14:01:23,895 chat.py:315: start_prompt 

```

### CODE REVIEW

File: src/instructlab/chat/chat.py [hunk 1]
```diff
@@ -475,6 +476,7 @@ def chat_cli(
     """Starts a CLI-based chat with the server"""
     cert = get_ssl_cert_config(tls_client_cert, tls_client_key, tls_client_passwd)
     verify = not tls_insecure
+    logger.debug("calling OpenAI API")
```
Comment (leseb) [inline src/instructlab/chat/chat.py:479]: I'm not sure whether this is a good usage of DEBUG logs. Also, this is not calling the API, we are building the client.

**Review by leseb — COMMENTED** [review]
I'm supportive of incorporating more DEBUG logs, but I'm somewhat hesitant about the additional logs added here. They don't appear to be particularly relevant, though it could just be my perspective. Let's wait to hear from other maintainers and get their input on this matter. Thanks for your efforts!

**Comment by russellb** [comment]
> I'm supportive of incorporating more DEBUG logs, but I'm somewhat hesitant about the additional logs added here. They don't appear to be particularly relevant, though it could just be my perspective. Let's wait to hear from other maintainers and get their input on this matter. Thanks for your efforts!

agreed

**Review by russellb — CHANGES_REQUESTED** [review]
i put feedback in a past comment but didn't use the official "review changes" bit to record it

**Pull request closed without merging**